chef-config/lib/chef-config/path_helper.rb

```diff
@@ -228,6 +228,17 @@ def self.all_homes(*args)
         joined_paths
       end
     end
+	
+    # On Windows, expands the given path so that it
+    # contains a drive letter at the beginning.
+    # [path] The path to convert
+    def self.get_full_windows_path(path)
+      if ChefConfig.windows?
+        path = cleanpath(File.absolute_path(path))
```

Comment:
Here's a chef recipe that shows why `::File.absolute_path` can cause problems -- if you run chef-client from a different directory, you get a different path. This would be very difficult to debug :smile: .

```ruby
log ::File.absolute_path('/a/b/c/d')
```
Here's the output:

% adamed@SEA-ADAMED1: 0x00000000: C:\Users\edwardsa
2015.06.23 Tue 06:03:42.03-07:00> chef-apply C:\test\ambiguous_path.rb
Recipe: (chef-apply cookbook)::(chef-apply recipe)
  * log[C:/a/b/c/d] action write

```sh
% C:\Users\edwardsa
> chef-apply C:\test\ambiguous_path.rb
Recipe: (chef-apply cookbook)::(chef-apply recipe)
  * log[C:/a/b/c/d] action write
% C:\Users\edwardsa
> z:
% Z:\adamedwards\hosted\chef-repo
> chef-apply C:\test\ambiguous_path.rb
Recipe: (chef-apply cookbook)::(chef-apply recipe)
  * log[Z:/a/b/c/d] action write
```
Notice that in one case, the path starts with `c:/` and in the other it starts with `z:/`.